Mathlib/Order/Bounds/Basic.lean

```diff
@@ -906,10 +906,20 @@ theorem IsGLB.exists_between' (h : IsGLB s a) (h' : a ∉ s) (hb : a < b) : ∃
 
 end LinearOrder
 
-theorem isGreatest_himp [GeneralizedHeytingAlgebra α] (a b : α) :
```

Comment:
I think these were better explicit.

Reply:
What's your argument for this? Mine for them being implicit is that they likely will never be used in a proof term that doesn't have enough context to prevent unification, and if ever that happens it would be better to add a type ascription rather than adding explicit arguments to some lemma in the middle of the proof term

Reply:
I can see this being the type of thing where I'd want to write `have := isGreatest_himp (foo x) (bar x)`, and then think a little using the goal state before proceeding. I don't think that `_`s within a proof term are likely to matter, especially since probably nothing much is going to use these lemmas.